hazelcast/src/test/java/com/hazelcast/internal/management/ManagementCenterServiceIntegrationTest.java

```diff
@@ -16,193 +16,143 @@
-                assertTrue(instance.node.clientEngine.isClientAllowed(client1));
-
-                Client client2 = new ClientImpl(null, createInetSocketAddress("127.0.0.2"), name, labels);
-                assertFalse(instance.node.clientEngine.isClientAllowed(client2));
-            }
-        });
+    public void testTimedMemberState_ignoresCache_longTimeFrame() {
+        String responseOne = mcs.getTimedMemberStateJson().orElse(null);
+        sleepSeconds(2);
+        String responseTwo = mcs.getTimedMemberStateJson().orElse(null);
+        assertNotSame(responseOne, responseTwo);
     }
 
     @Test
-    public void testMemberStateNameProvided() {
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                String responseString = doHttpGet("/mancen/memberStateCheck");
-                assertNotNull(responseString);
-                assertNotEquals("", responseString);
-
-                JsonObject object;
-                try {
-                    object = Json.parse(responseString).asObject();
-                } catch (ParseException e) {
-                    throw new AssertionError("Failed to parse JSON: " + responseString);
-                }
-                TimedMemberState memberState = new TimedMemberState();
-                memberState.fromJson(object);
-                assertEquals(instance.getName(), memberState.getMemberState().getName());
-            }
-        });
+    public void testMCEvents_storesEvents_recentPoll() {
+        mcs.pollMCEvents();
+
+        TestEvent expectedEvent = new TestEvent();
+        mcs.log(expectedEvent);
+        mcs.log(new TestEvent());
+        mcs.log(new TestEvent());
+
+        List<Event> actualEvents = mcs.pollMCEvents();
+        assertEquals(3, actualEvents.size());
+        assertEquals(expectedEvent, actualEvents.get(0));
     }
 
-    private InetSocketAddress createInetSocketAddress(String name) {
-        try {
-            return new InetSocketAddress(InetAddress.getByName(name), 5000);
-        } catch (UnknownHostException e) {
-            throw new RuntimeException(e);
-        }
+    @Test
+    public void testMCEvents_ignoresEvents_noPollAtAll() {
+        mcs.log(new TestEvent());
+        mcs.log(new TestEvent());
+
+        assertEquals(0, mcs.pollMCEvents().size());
     }
 
-    private static int availablePort() {
-        while (true) {
-            int port = 8000 + (int) (1000 * Math.random());
-            try {
-                ServerSocket socket = new ServerSocket(port);
-                socket.close();
-                return port;
-            } catch (Exception ignore) {
-                // try next port
-            }
-        }
+    @Test
+    public void testMCEvents_clearsEventQueue_noRecentPoll() {
+        mcs.pollMCEvents();
+
+        mcs.log(new TestEvent());
+        mcs.log(new TestEvent());
+
+        sleepSeconds(31);
```

Comment:
Maybe we can add a `onMcEventWindowExceeded()` or a similar method and make it package-private that the prod code and the test can call instead of sleeping for a half-second.

Reply:
I was thinking of overriding timeouts for test purposes, but a package-private method will work as well.

Thanks for the suggestion. Going to address it.